src/render/opengl/SDL_render_gl.c

```diff
@@ -1961,11 +1994,15 @@ SDL_RenderDriver GL_RenderDriver = {
     GL_CreateRenderer,
     { "opengl",
       (SDL_RENDERER_ACCELERATED | SDL_RENDERER_PRESENTVSYNC | SDL_RENDERER_TARGETTEXTURE),
-      4,
-      { SDL_PIXELFORMAT_ARGB8888,
-        SDL_PIXELFORMAT_ABGR8888,
-        SDL_PIXELFORMAT_RGB888,
-        SDL_PIXELFORMAT_BGR888 },
+      8,
+      { SDL_PIXELFORMAT_RGBA32,
+        SDL_PIXELFORMAT_RGBX32,
+        SDL_PIXELFORMAT_RGB24,
+        SDL_PIXELFORMAT_ARGB8888,
+        SDL_PIXELFORMAT_XRGB8888,
+        SDL_PIXELFORMAT_RGB565,
```

Comment:
Which OpenGL version is required by the render API? Given that before this PR it was using `GL_UNSIGNED_INT_8_8_8_8_REV` for all RGB pixel formats, I'd assume that it requires OpenGL 1.2 or later and therefore the other formats are also available.

Supporting OpenGL 1.1 would be nice, but it might be better to defer it to a separate PR since I wouldn't be able to properly test it.

Reply:
Full 565 support (which is still emulated in drivers) is part of https://registry.khronos.org/OpenGL/extensions/ARB/ARB_ES2_compatibility.txt , which became core in OpenGL 4.1.

Reply:
Additionally, if you look at the Required Texture Formats section of the OpenGL 3.3 specification (https://registry.khronos.org/OpenGL/specs/gl/glspec33.core.pdf), the packed 16 bit formats are not listed there.